packages/admin/cms-admin/src/generator/future/generateForm/generateFormField.ts

```diff
@@ -1,17 +1,37 @@
 import { IntrospectionEnumType, IntrospectionInputValue, IntrospectionNamedTypeRef, IntrospectionObjectType, IntrospectionQuery } from "graphql";
 
+import { Prop } from "../generateForm";
 import { FormConfig, FormFieldConfig, isFormFieldConfig } from "../generator";
 import { camelCaseToHumanReadable } from "../utils/camelCaseToHumanReadable";
 import { findQueryTypeOrThrow } from "../utils/findQueryType";
 import { Imports } from "../utils/generateImportsCode";
 import { isFieldOptional } from "../utils/isFieldOptional";
 import { findFieldByName, GenerateFieldsReturn } from "./generateFields";
 
-function getTypeInfo(arg: IntrospectionInputValue, gqlIntrospection: IntrospectionQuery) {
+function convertGqlScalarToTypescript(scalarName: string) {
+    if (scalarName === "String" || scalarName === "ID") {
+        return "string";
+    } else if (scalarName === "Boolean") {
+        return "boolean";
+    } else if (scalarName === "Int" || scalarName === "Float") {
+        return "number";
+    } else if (scalarName === "DateTime") {
+        return "Date";
```

Comment:
I don't think that's correct. This would probably be string, depending on the codegen config. We could use the `Scalars` type from `src/graphql.generated.ts` 🤔

Reply:
That's what I've thought of. @nsams what do you think?

Reply:
yes, sounds good.

But I doubt you would write in handmade `Scalars.DateTime`, instead you would just use `string` or `Date` (whatever is configured).

An alternative could be to analyze the graphql.generated.ts file or codegen.ts file during generation-time and generate the resulting type. (Of course that is much more difficult to implement)

Reply:
Maybe we should simply use string for DateTime, as our date fields all work with strings since v7?

Reply:
> as our date fields all work with strings since v7

ah, so for v7+ the codegen config needs to define Date/DateTime as string? Then we can hardcode here string as well.

Reply:
DateTime is already defined as string: https://github.com/vivid-planet/comet-starter/blob/main/site/codegen.ts#L15. We should check how Nest handles `Date`s sent via GraphQL, but I'm pretty sure they're using a string representation by default.

Reply:
> We should check how Nest handles Dates sent via GraphQL

you can only send JSON - which does not include Date objects

Reply:
@andrearutrecht and @Ben-Ho used a middleware for date fields, so it would be interesting why they did it.

```ts
@Field(() => Date, {
  nullable: true,
  middleware: [
    async (ctx: MiddlewareContext, next: NextFn) => {
      const value = await next();
      return value ? new Date(value) : null;
    },
  ],
})
birthDate: string | null;
```

Reply:
@johnnyomair I just wanted to use the correct typing. I don't want care about how graphql handles dates and what format is required. If it's a date it should be a date until serialization, which is not my responsibility but of nestjs/graphql-module. same client-side.

Reply:
Additionally a more detailed graphql-schema does provide more options for admin-generator. because if a field is string it's different than Scalars.DateTime, even if they are basically both strings. Same for some other custom string-format-type.

Reply:
> I just wanted to use the correct typing. I don't want care about how graphql handles dates and what format is required. If it's a date it should be a date until serialization, which is not my responsibility but of nestjs/graphql-module. same client-side.

I'm not sure about this. I'm thinking about date-only dates, where a real Date could lead to problems (timezone issues, daylight saving time), whereas a string `"2024-10-15"` doesn't.

> Additionally a more detailed graphql-schema does provide more options for admin-generator. because if a field is string it's different than Scalars.DateTime, even if they are basically both strings. Same for some other custom string-format-type.

So you're saying we should use the Scalars type all the time?

Reply:
We [decided](https://github.com/vivid-planet/comet/pull/1916) NOT to work with Date Objects in DatePicker, instead we use strings. For the reasons @johnnyomair mentioned.

Imho you should do the same in your project: convert to a Date object as late as possible (or never)